public/app/features/alerting/unified/components/rule-editor/TemplateStep/TemplateStep.tsx

```diff
@@ -79,7 +79,9 @@ export const TemplateStep: FC = () => {
 
   const setRuleNameAfterTemplate = useCallback(
     (template?: Template) => {
-      if (!getValues('ruleName')) {
+      const value = getValues('ruleName');
+      const valueExists = templates.current.find((opt) => value === `${opt.name} Alerting Rule`);
```

Comment:
Not sure I follow this one. `templates.current` hold the templates from the API, right? Why are we looking for a template with this name?

Reply:
Because the goal is to change names only until user adds a custom name, then it should stay the same. So I am checking if value in ruleName field is the same as one of the templates names or if user changed it.

Reply:
Oh, I think I get it now, thanks 👍